core/src/test/java/org/apache/iceberg/rest/TestRESTCatalog.java

```diff
@@ -3311,6 +3312,92 @@ public void testClientDoesNotSendIdempotencyWhenServerNotAdvertising() {
     local.dropTable(ident);
   }
 
+  @Test
+  public void testClientWithDefaultNamespaceSeparator() {
```

Comment:
```suggestion
  public void clientWithLegacyNamespaceSeparator() {
```

Reply:
we already have a mix of using `test` prefixes vs not, and using `test` prefixes for new stuff doesn't really add any value